### DISCUSSION TITLE
Setting and accessing PORT in cli in dev env in next@9.3.0

### DISCUSSION THREAD

**Discussion by steveswork** [vercel/next.js#12504] [Help] [answered]
How can I access the server port number? The default next js port is 3000. This port port does not exist in process.env object. It also does not exist in process.dev when I attempt to set it either by running "next -p 3000" or "cross-env PORT=3000 && next dev"

Background: I am getting _"127.0.0.1:80 ECONNREFUSED"_ error when running api calls in the server. I do not know where PORT 80 is coming from: the server port is 3000. My workaround idea is to use the absolute URL. But the process.env.PORT is undefined.

As per @leerob request, following is a relevant snippet of the app architecture provided. Api fetch modules are placed in a single utils folder.  All remote api requests like those from Redux actions and getInitialProps are made to the fetch modules. Fetch module constructs `AxiosRequestConfig` config object based on the params submitted. I think that this is part that causes the ECONNREFUSED error during api fetch in the server. I would like to set the axios config baseURL property manually using env calibrated means. 

The idea is to add `getBaseUrl()` to browser and server properties where `browser.getBaseUrl(): location.origin` while `server.getBaseUrl():  '/${ process.env.PORT }'`.  Considering that the browser and server properties are modules located in `properties` folder which contains an `index` module that requires the proper properties module depending on whether the build is for the browser or for the server. The axios baseURL property could then be set to properties.getBaseUrl().

**Comment by leerob** [comment]
Could you share an example on CodeSandbox or explain your app architecture more, maybe?

**Reply by steveswork** [reply]
Hi @leerob, sure. I have now updated my post to provide further explanation.

**Comment by steveswork** [comment, marked as answer]
The best I can come up with at this time is: if `server.getBaseUrl()` does not exist, create it using the port info pulled from current server request headers either from the `pages/_document.js` `static Document.getInitialProps'` `context.req` arg or from the `pages/api` routes' `req` arg.

Hope there is a way to intercept all requests to `pages/api` routes in one place with a middleware call to streamline the process. Or perhaps, I am currently unaware of it. At the moment, I am using next-connect in all `pages/api` routes to attach the middleware that pulls the port info.
